client/src/views/uploadPage.vue

```diff
@@ -78,27 +81,72 @@ async function onCropperReady() {
 		console.log(error)
 	}
 }
+
+//checks bytes of file header to check file type because human readable MIME type can be manipulated
+function checkFileType(file) {
+  const reader = new FileReader();
+  let header = "";
+
+  reader.onload = (e) => {
+    let fileType = "";
+    let arr = new Uint8Array(e.target.result).subarray(0, 4);
+
+    for (let i = 0; i < arr.length; i++) {
+      header += arr[i].toString(16);
+    }
+
+    //hexadecimal representation of those file extensions references: https://mimesniff.spec.whatwg.org/#matching-an-image-type-pattern
+    //https://en.wikipedia.org/wiki/List_of_file_signatures
+    if (header.includes("424d")) {
+      fileType = "bmp";
+    } else if (header.includes("ffd8ff")) {
+      fileType = "jpg";
+    } else if (header.includes("504e47")) {
+      fileType = "png";
+    } else if (header.includes("57454250")) {
```

Comment:
you are only checking first 4 bytes here, which according to [google's specification doc](https://developers.google.com/speed/webp/docs/riff_container#webp_file_header), should be `RIFF` instead of `WEBP`.
Consider change this line to following so you can check both `RIFF` & `WEBP` headers:

```js
else if (header.includes("52494646") && header.includes("57454250"))
```